### PR TITLE
Add sync APIs for BlobValue

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/BlobValue.cs [hunk 1]
```diff
@@ -29,22 +29,46 @@ public override void Visit(IValueVisitor visitor)
 
         public override object ToObject() => this;
 
+        public bool IsString => Content.IsString;
+
+        public bool IsBase64 => Content.IsBase64;
+
+        public bool IsByteArray => Content.IsByteArray;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/BlobValue.cs:36]: We need clear invariants on what *exactly* these mean. 

For example, base64 is also a string. 

And byte contents can still be accessed as either Byte or Base64.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Functions/Blob/Base64Blob.cs [hunk 2]
```diff
@@ -25,10 +25,26 @@ internal Base64Blob(string base64Str)
             }
         }
 
+        internal override bool IsString => false;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Functions/Blob/Base64Blob.cs:28]: >IsString => false; [](http://example.com/codeflow?start=31&length=18)

why not true?
Base64 are strings. 

What's the rule we're following here.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Functions/Blob/Base64Blob.cs:28]: Chatted offline. What customer really wants is #2220 instead.

**Pull request closed without merging**